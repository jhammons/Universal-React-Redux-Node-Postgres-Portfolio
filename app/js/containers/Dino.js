import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import * as DinoActions from '../actions/dino';

// @connect(state => { users: state.users })
class Dino extends Component {

    static readyOnActions(dispatch) {
        return dispatch(DinoActions.dinoInit(6));
    }
    
    componentDidMount() {
       Dino.readyOnActions(this.props.dispatch);
    }

    handleClick(e) {
        e.preventDefault();
        this.props.dispatch(DinoActions.dinoGrowled(1));
    }

    render() {

        const {growls} = this.props.dino;

        return (
            <div>
                <Helmet title='Dino' />
                <h5>Dinosaur Page</h5>
                <p>Hello dino. You have growled {growls} times!</p>
                <div><button onClick={this.handleClick.bind(this)}>Growl Again!</button></div>
                <p>Now you can go <Link to="/">home!</Link></p>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        dino: state.dino
    };
}

export default connect(mapStateToProps)(Dino);
