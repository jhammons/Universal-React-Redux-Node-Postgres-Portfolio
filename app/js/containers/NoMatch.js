import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { Link } from 'react-router';

class NoMatch extends Component {

  render() {
    return (
      <div className="center-container">
        <Helmet title='404 Not Found' htmlAttributes={{"class": "not-found-page"}} />
          <div className="not-found-404">
              <img src="../../images/404.png" alt="Not Found"/>
              <Link to="/"><button className="button button-green button-smaller">Go Home</button></Link>
          </div>
      </div>
    )
  }
}

export default NoMatch;
