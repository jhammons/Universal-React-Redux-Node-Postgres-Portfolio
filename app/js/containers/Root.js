import React, {Component} from 'react';

class Root extends Component {

    renderInitialState() {
        if (this.props.initialState) {
            const innerHtml = `window.__INITIAL_STATE__ = ${JSON.stringify(this.props.initialState)}`;
            return <script dangerouslySetInnerHTML={{__html: innerHtml}}/>;
        }
    }

    renderEnvironment() {
        const innerHtml = `window.__ENVIRONMENT__ = '${__ENVIRONMENT__}'`;
        return <script dangerouslySetInnerHTML={{__html: innerHtml}}/>
    }

    render() {
        const head = this.props.head;
        const attrs = head.htmlAttributes.toComponent();

        return (
            <html {...attrs}>
            <head>
                {head.title.toComponent()}
                {head.meta.toComponent()}
                <link rel="apple-touch-icon" sizes="180x180" href="/images/favicons/apple-touch-icon.png"/>
                <link rel="icon" type="image/png" href="/images/favicons/favicon-32x32.png" sizes="32x32"/>
                <link rel="icon" type="image/png" href="/images/favicons/android-chrome-192x192.png" sizes="192x192"/>
                <link rel="icon" type="image/png" href="/images/favicons/favicon-16x16.png" sizes="16x16"/>
                <link rel="manifest" href="/images/favicons/manifest.json"/>
                <link rel="mask-icon" href="/images/favicons/safari-pinned-tab.svg" color="#5bbad5"/>
                <link rel="shortcut icon" href="/images/favicons/favicon.ico"/>
                <meta name="msapplication-config" content="/images/favicons/browserconfig.xml"/>
                <meta name="theme-color" content="#ffffff"/>
                {head.link.toComponent()}
            </head>
            <body>
            <div id='root' dangerouslySetInnerHTML={{__html: this.props.content}}/>
            {this.renderEnvironment()}
            {this.renderInitialState()}
            {head.script.toComponent()}
            <script src={!process.env.NODE_ENV ? 'js/global.js' : '/js/global.min.js'}></script>
            </body>
            </html>
    );
    }
    }

    export default Root;
