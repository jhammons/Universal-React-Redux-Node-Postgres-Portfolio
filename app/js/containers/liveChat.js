import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import LiveChat from '../components/live-chat/liveChat';
import * as chatActions from '../actions/liveChat';
import io from 'socket.io-client';

if (typeof window !== 'undefined') require('../../scss/live-chat/live-chat.scss');

class LiveChatContainer extends Component {
    //
    // static readyOnActions(dispatch) {
    //     return Promise.all([
    //         dispatch(chatActions.initSockets())
    //     ]);
    // }

    constructor(props) {
        super(props);
        this.socketChatmessage = this.socketChatmessage.bind(this);
        this.socketTyping = this.socketTyping.bind(this);
        this.socketDisconnect = this.socketDisconnect.bind(this);
    }

    componentDidMount() {
        this.socket = io('http://localhost:3100/live-chat');
        this.socket.on('connect', () => {
            console.log("socket.io connected. Id: " + this.socket.id);
        });
        this.socket.on('chatMessage', this.socketChatmessage);
        this.socket.on('typing', this.socketTyping);
        this.socket.on('disconnect', this.socketDisconnect);
        this.socket.on('status', this.socketStatus);
    }

    componentWillUnmount() {
        clearTimeout(this.typingTimer);
    }

    socketStatus(message) {
        console.log(message);
    }

    socketChatmessage(message) {
        if (message.id !== this.props.liveChat.room) {
            // console.log(this.typingTimer);
            clearTimeout(this.typingTimer);
            this.props.dispatch(chatActions.chatIsTyping(false));
            this.props.dispatch(chatActions.chatSetRemoteId(message.id, message.name));
        }
        this.props.dispatch(chatActions.chatReceiveMesssage(message));
        console.log("transport: " + this.socket.io.engine.transport.name);

    }

    socketTyping(id) {
        // const parsedId = id.slice(id.indexOf('#') +1);
        if (id === this.props.liveChat.remoteId) {
            clearTimeout(this.typingTimer);
            this.props.dispatch(chatActions.chatIsTyping(true));
            this.typingDelay();
        }
    }

    socketDisconnect(id) {

        console.log(id);
        // if (id === this.state.id) {
        //     alert('you\'ve been disconnected!');
        // } else {
        //     this.setState({
        //         messages: this.state.messages.map(m => {
        //             return m.id === id ?
        //                 Object.assign(m, {connected: false}) : m;
        //         })
        //     });
        // }
    }

    typingDelay() {
        return this.typingTimer = setTimeout( () => {
            this.props.dispatch(chatActions.chatIsTyping(false));
        }, 1200);
    }

    onSubmit(e) {
        e.preventDefault();
        let chatInput = e.currentTarget[0].value.trim();
        if (!chatInput) return;
        const message = {
            id: this.props.liveChat.room,
            room: this.props.liveChat.room,
            name: this.props.liveChat.localName,
            message: chatInput,
            date: Date.now()
        };
        this.socket.emit('chatMessage', message);
        this.props.dispatch(chatActions.chatCreateMesssage(''));
        console.log("new message: ", message);
    }

    onChange(e) {
        this.socket.emit('typing', this.socket.id);
        this.props.dispatch(chatActions.chatOnChange(e.target.value));
    }

    newUser(e) {
        e.preventDefault();
        const name = e.currentTarget[0].value.trim();
        if (!name) return;

        this.socket.emit('newUser', {
            id: this.socket.id,
            name: name,
            connected: true
        }, (room) => {
            this.props.dispatch(chatActions.chatNewUser({
                room: room,
                name: name,
                registered: true
            }));
        });
    }

    render() {
        return (
            <div>
                <LiveChat onSubmit={this.onSubmit.bind(this)}
                          onChange={this.onChange.bind(this)}
                          newUser={this.newUser.bind(this)}
                          chatOpen={this.props.liveChat.chatOpen}
                          dispatch={this.props.dispatch}
                          {...this.props.liveChat} />
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        liveChat: state.liveChat
    };
}
function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LiveChatContainer);
