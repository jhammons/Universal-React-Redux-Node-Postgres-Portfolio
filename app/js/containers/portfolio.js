import React, {Component} from 'react';
import {connect} from 'react-redux';
import * as portfolioActions  from '../actions/portfolio';
import PortfolioItem from '../components/portfolioItem';

export default class Portfolio extends React.Component {

    static readyOnActions(dispatch) {
        return Promise.all([
            dispatch(portfolioActions.fetchPortfolioIfNeeded())
        ]);
    }

    componentDidMount() {
        Portfolio.readyOnActions(this.props.dispatch);
    }

    renderPortfolio() {
        if (this.props.portfolio.readyState === 'WORK_FETCHING') {
            return 'fetching items...';
        } else if (this.props.portfolio.readyState === 'WORK_FETCHED') {
            return <div>
                {
                    this.props.portfolio.items.map((item, index)=> {
                        let color = (index % 2) ? '#000' : '#353535';
                        return  <PortfolioItem key={index} color={color} item={item} />
                    })
                }
            </div>
        } else {
            return 'nothing found.'
        }
    }
    
    render() {
        return (
            <section className="portfolio clearfix" id="work">
                {this.renderPortfolio()}
            </section>
        );
    }
}

function mapStateToProps(state) {
    return {
        portfolio: state.portfolio
    };
}

export default connect(mapStateToProps)(Portfolio);