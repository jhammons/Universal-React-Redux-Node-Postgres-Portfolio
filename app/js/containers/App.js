import React, {Component} from 'react';
import Helmet from 'react-helmet';
import Footer from '../components/common/footer';
import LiveChatContainer from './liveChat';
import LiveChatLauncher from '../components/live-chat/chatLauncher';
import {toggleChat} from '../actions/liveChat';
import { connect } from 'react-redux';

if (typeof document !== 'undefined') require('../../scss/global.scss');

export class App extends Component {

    closeChat() {
        if (this.props.liveChat.chatOpen) {
            this.props.dispatch(toggleChat(false));
        }
    }

    render() {
        
        let stylesheets = [
            {'rel': 'stylesheet', 'href': 'https://fonts.googleapis.com/css?family=Oswald|Sanchez', 'async': true}
        ];

        if (process.env.NODE_ENV === 'production') stylesheets.unshift({
            'rel': 'stylesheet',
            'href': '/css/global.min.css'
        });

        // footer needs a different background color on homepage vs. the rest of site
        // There must be a better way...
        let footer = null;
        if (
            typeof this.props.children !== 'undefined' &&
            typeof this.props.children.props !== 'undefined' &&
            typeof this.props.children.props.route !== 'undefined' &&
            typeof this.props.children.props.route.path !== 'undefined'
        ) {
            footer = this.props.children.props.route.path === '*' 
                ? null
                : <Footer slug={this.props.children.props.route.path.slice(1)}/>;
        }

        return (
            <div className="page-wrapper">
                <div className={this.props.liveChat.chatOpen ? 'content-wrapper live-chat-open' : 'content-wrapper'}
                     onClick={this.closeChat.bind(this)}>
                    <Helmet
                        title='Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal'
                        titleTemplate='Jason Gallagher - %s'
                        meta={[
                            {'char-set': 'utf-8'},
                            {
                                'name': 'description',
                                'content': 'Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.'
                            },
                            {'name': 'viewport', 'content': 'width=device-width, initial-scale=1'}
                        ]}
                        link={stylesheets}
                    />
                    {this.props.children}
                    {footer}
                </div>
                <LiveChatLauncher />
                <LiveChatContainer />
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        liveChat: state.liveChat
    };
}

function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
