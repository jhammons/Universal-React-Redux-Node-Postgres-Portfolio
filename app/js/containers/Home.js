import React, {Component} from 'react';
import {connect} from 'react-redux';
import Helmet from 'react-helmet';
import StickyHeader from '../components/common/stickyHeader';
import ResponsiveMenu from '../components/common/responsiveMenu';
import Header from '../components/home/header';
import WhoWhatWhere from '../components/home/whoWhatWhere';
import About from '../components/home/about';
import Portfolio from './Portfolio';
import * as portfolioActions  from '../actions/portfolio';
import * as globalActions  from '../actions/global';
import { animation, getEnvironment } from '../libs/utils';
import throttle from 'lodash/throttle';

if (typeof document !== 'undefined') require('../../scss/home/home.scss');

class Home extends Component {

    static readyOnActions(dispatch) {
        return Promise.all([
            dispatch(portfolioActions.fetchPortfolioIfNeeded())
        ]);
    }

    constructor(props) {
        super(props);
        this.boundHandleScroll = throttle(this.handleScroll.bind(this, props.dispatch), 150);
    }

    componentDidMount() {
        Home.readyOnActions(this.props.dispatch)
            .then(() => {
                if (getEnvironment('client') && (window.pageYOffset > 10 || window.location.hash)) {
                    this.props.dispatch(globalActions.animateOff());
                }
                getEnvironment('client') && window.addEventListener('scroll', this.boundHandleScroll);
            });
    }

    componentWillUnmount() {
        window.removeEventListener('scroll', this.boundHandleScroll);
        document.documentElement.className = '';
        const {dispatch, global} = this.props;
        if (global.headerState) dispatch(globalActions.setHeader(false));
    }

    componentDidUpdate() {
        if (getEnvironment('client') && !this.props.global.animateOff && window.location.hash) {
            setTimeout(() => {
                this.props.dispatch(globalActions.animateOff());
            }, 0);
        }
    }

    handleScroll(dispatch) {
        const scrl = window.pageYOffset;

        // handle sticky header
        if (scrl >= 250 && this.props.global.headerState === false) {
            document.documentElement.className = 'sticky-menu-open';
            dispatch(globalActions.setHeader(true));
        } else if (scrl < 250 && this.props.global.headerState === true) {
            document.documentElement.className = '';
            dispatch(globalActions.setHeader(false));
        }

        // handle scroll animations
        // elem = id of element to animate, action = action to dispatch
        function setupAnimation(elem, action, props) {
            const viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight) || 0;

            if (!props.global[action] &&
                document.getElementById(elem).getBoundingClientRect().top < viewportHeight * 0.9) {
                dispatch(globalActions[action](true));
            }
        }

        setupAnimation('about', 'animateAbout', this.props);
        setupAnimation('work', 'animatePortfolio', this.props);
        setupAnimation('contact-area', 'animateContact', this.props);
    }
    
    openBurger() {
        this.props.dispatch(globalActions.openHamburger());
    }

    closeBurger() {
        this.props.dispatch(globalActions.closeHamburger());
    }

    render() {
        
        const {
            headerState,
            hamburgerState,
            htmlClass,
            animateOff,
            animateAbout,
            animatePortfolio
        } = this.props.global;

        const htmlClassCheck = htmlClass ? {"class": htmlClass} : {};

        return (
            <div id="home">
                <Helmet
                    htmlAttributes={htmlClassCheck}
                    link = {[
                        {"rel": "canonical", "href": "https://jasongallagher.org"},
                    ]}
                />
                <StickyHeader visible={headerState} displayHamburger={false}/>
                <ResponsiveMenu closeBurger={this.closeBurger.bind(this)}
                        menuVisible={hamburgerState}
                        />
                <Header openBurger={this.openBurger.bind(this)} />
                <WhoWhatWhere />
                <About animation={animation.bind(this, animateAbout, animateOff)} />
                <Portfolio animation={animation.bind(this, animatePortfolio, animateOff)} />
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        portfolio: state.portfolio,
        global: state.global
    };
}
function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);