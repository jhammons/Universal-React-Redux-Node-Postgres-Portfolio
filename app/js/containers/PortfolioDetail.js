import React, {Component} from 'react';
import { browserHistory } from 'react-router'
import {connect} from 'react-redux';
import Helmet from 'react-helmet';
import StickyHeader from '../components/common/stickyHeader';
import Loader from '../components/common/loader';
import ResponsiveMenu from '../components/common/responsiveMenu';
import * as portfolioActions  from '../actions/portfolio';
import * as globalActions  from '../actions/global';

if (typeof document !== 'undefined') require('../../scss/portfolio-detail/portfolio_detail.scss');

class PortfolioDetail extends Component {

    static readyOnActions(dispatch, params, bypassCheck = false) {
        return Promise.all([
            dispatch(portfolioActions.fetchPortfolioDetailIfNeeded(params.slug, bypassCheck)),
            dispatch(globalActions.animateOff())
        ]);
    }

    componentDidMount() {
        const {dispatch, params} = this.props;
        PortfolioDetail.readyOnActions(dispatch, params);
        window.scrollTo(0, 0);
    }

    componentWillUnmount() {
        const {dispatch} = this.props;
        dispatch(portfolioActions.resetPortfolioDetail());
    }

    componentWillReceiveProps(nextProps) {
        const {DetailReadyState} = this.props.portfolio;

        if (nextProps.params.slug !== this.props.params.slug
            && DetailReadyState === 'WORK_DETAIL_FETCHED') {

            // get height from project for next spinner so it doesn't bounce
            const lastProject = this.refs.lastProject.clientHeight;
            this.props.dispatch(portfolioActions.lastProjectHeight(lastProject));

            const {dispatch, params} = nextProps;
            PortfolioDetail.readyOnActions(dispatch, params, true);
            window.scrollTo(0, 0);
        }
    }

    openBurger() {
        this.props.dispatch(globalActions.openHamburger());
    }
    
    closeBurger() {
        this.props.dispatch(globalActions.closeHamburger());
    }

    renderPortfolioItemDetail(browserHistory) {
        const {DetailReadyState, lastProjectHeight, item} = this.props.portfolio;

        if (DetailReadyState === 'WORK_DETAIL_FETCHING') {

            return <Loader height={lastProjectHeight} />;

        } else if (DetailReadyState === 'WORK_DETAIL_FETCHED') {

            return (
                <div ref="lastProject">
                    {this.props.children}
                </div>
            );

        } else if (DetailReadyState === 'WORK_DETAIL_FETCH_FAILED') {
            browserHistory.push('/not-found');
        }
    }

    render() {

        const { hamburgerState, htmlClass } = this.props.global;
        const { item } = this.props.portfolio;

        const title = (typeof window === 'undefined') ? item.title : this.props.params.slug;
        const url_slug = (typeof window === 'undefined') ? item.url_slug : this.props.params.slug;
        const metadesc = item ? item.description.slice(0, 300) : ''; // TODO: item.description sometimes undefined

        const htmlClassCheck = htmlClass ? {"class": htmlClass} : {};

        // hide header on small devices when chat is open
        const headerVisible = () => {
            if (typeof window !== 'undefined') {
                return window.innerWidth <= 768 ? !this.props.liveChat.chatOpen : true;
            } else return true;
        };

        return (
            <div id="interior-page">
                <Helmet
                    title={title}
                    htmlAttributes={htmlClassCheck}
                    meta={[
                        {'name': 'description', 'content': metadesc}
                    ]}
                    link = {[
                        {"rel": "canonical", "href": "https://jasongallagher.org/portfolio/" + url_slug}
                    ]}
                />
                <StickyHeader visible={headerVisible()}
                              displayHamburger={true}
                              openBurger={this.openBurger.bind(this)}
                              remoteId={this.props.liveChat.remoteId}
                              dispatch={this.props.dispatch}
                />
                <ResponsiveMenu
                    closeBurger={this.closeBurger.bind(this)}
                    menuVisible={hamburgerState}
                />

                {this.renderPortfolioItemDetail(browserHistory)}
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        portfolio: state.portfolio,
        global: state.global,
        contactForm: state.contactForm,
        liveChat: state.liveChat
    };
}
function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(PortfolioDetail);