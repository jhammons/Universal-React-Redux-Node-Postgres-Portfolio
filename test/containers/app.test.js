import { App } from '../../app/js/containers/App';
import Helmet from 'react-helmet';
import Footer from '../../app/js/components/common/footer';
import LiveChatContainer from '../../app/js/containers/liveChat';
import LiveChatLauncher from '../../app/js/components/live-chat/chatLauncher';

describe('<App />', () => {
    let wrapper;
    const liveChat = {
        chatOpen: false
    };

    beforeEach(() => {
        wrapper = shallow(<App liveChat={liveChat} />);
    });

    it('the app renders...', () => {
        expect(wrapper.find('.page-wrapper')).toHaveLength(1);
    });

    it('React Helmet is used', () => {
        expect(wrapper.find(Helmet)).toHaveLength(1);
    });

    it('renders <LiveChatContainer />', () => {
        expect(wrapper.find(LiveChatContainer)).toHaveLength(1);
    });

    it('renders <LiveChatLauncher />', () => {
        expect(wrapper.find(LiveChatLauncher)).toHaveLength(1);
    });

    it('renders <Footer /> on standard routes', () => {
        const children = {
            props: {
                route: {
                    path: '/index'
                }
            }
        };

        wrapper = shallow(<App liveChat={liveChat} children={children} />);
        expect(wrapper.find(Footer)).toHaveLength(1);
    });

    it.skip('matches snapshot (without footer)', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
