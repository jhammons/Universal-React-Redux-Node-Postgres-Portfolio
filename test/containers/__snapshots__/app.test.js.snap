// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot (without footer) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    liveChat={
      Object {
        "chatOpen": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="content-wrapper"
          onClick={[Function]}
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "/css/global.min.css",
                  "rel": "stylesheet",
                },
                Object {
                  "async": true,
                  "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                  "rel": "stylesheet",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "char-set": "utf-8",
                },
                Object {
                  "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                  "name": "description",
                },
                Object {
                  "content": "width=device-width, initial-scale=1",
                  "name": "viewport",
                },
              ]
            }
            title="Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal"
            titleTemplate="Jason Gallagher - %s"
          />
        </div>,
        <Connect(LiveChatLauncher) />,
        <Connect(LiveChatContainer) />,
      ],
      "className": "page-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "/css/global.min.css",
                    "rel": "stylesheet",
                  },
                  Object {
                    "async": true,
                    "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                    "rel": "stylesheet",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "char-set": "utf-8",
                  },
                  Object {
                    "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                    "name": "description",
                  },
                  Object {
                    "content": "width=device-width, initial-scale=1",
                    "name": "viewport",
                  },
                ]
              }
              title="Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal"
              titleTemplate="Jason Gallagher - %s"
            />,
            undefined,
            null,
          ],
          "className": "content-wrapper",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defer": true,
              "encodeSpecialCharacters": true,
              "link": Array [
                Object {
                  "href": "/css/global.min.css",
                  "rel": "stylesheet",
                },
                Object {
                  "async": true,
                  "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                  "rel": "stylesheet",
                },
              ],
              "meta": Array [
                Object {
                  "char-set": "utf-8",
                },
                Object {
                  "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                  "name": "description",
                },
                Object {
                  "content": "width=device-width, initial-scale=1",
                  "name": "viewport",
                },
              ],
              "title": "Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal",
              "titleTemplate": "Jason Gallagher - %s",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="content-wrapper"
            onClick={[Function]}
          >
            <HelmetWrapper
              defer={true}
              encodeSpecialCharacters={true}
              link={
                Array [
                  Object {
                    "href": "/css/global.min.css",
                    "rel": "stylesheet",
                  },
                  Object {
                    "async": true,
                    "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                    "rel": "stylesheet",
                  },
                ]
              }
              meta={
                Array [
                  Object {
                    "char-set": "utf-8",
                  },
                  Object {
                    "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                    "name": "description",
                  },
                  Object {
                    "content": "width=device-width, initial-scale=1",
                    "name": "viewport",
                  },
                ]
              }
              title="Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal"
              titleTemplate="Jason Gallagher - %s"
            />
          </div>,
          <Connect(LiveChatLauncher) />,
          <Connect(LiveChatContainer) />,
        ],
        "className": "page-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HelmetWrapper
                defer={true}
                encodeSpecialCharacters={true}
                link={
                  Array [
                    Object {
                      "href": "/css/global.min.css",
                      "rel": "stylesheet",
                    },
                    Object {
                      "async": true,
                      "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                      "rel": "stylesheet",
                    },
                  ]
                }
                meta={
                  Array [
                    Object {
                      "char-set": "utf-8",
                    },
                    Object {
                      "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                      "name": "description",
                    },
                    Object {
                      "content": "width=device-width, initial-scale=1",
                      "name": "viewport",
                    },
                  ]
                }
                title="Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal"
                titleTemplate="Jason Gallagher - %s"
              />,
              undefined,
              null,
            ],
            "className": "content-wrapper",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defer": true,
                "encodeSpecialCharacters": true,
                "link": Array [
                  Object {
                    "href": "/css/global.min.css",
                    "rel": "stylesheet",
                  },
                  Object {
                    "async": true,
                    "href": "https://fonts.googleapis.com/css?family=Oswald|Sanchez",
                    "rel": "stylesheet",
                  },
                ],
                "meta": Array [
                  Object {
                    "char-set": "utf-8",
                  },
                  Object {
                    "content": "Full Stack Engineer in Berkeley. Focused on React, Angular, Node.Js, Drupal, UI/UX. San Francisco Bay Area, East Bay, San Francisco and Oakland.",
                    "name": "description",
                  },
                  Object {
                    "content": "width=device-width, initial-scale=1",
                    "name": "viewport",
                  },
                ],
                "title": "Front End Engineer in Berkeley, CA specializing in React, Javascript, Node.Js, Angular and Drupal",
                "titleTemplate": "Jason Gallagher - %s",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
